File: codeforces/1600/1648/A.cpp
#include<iostream>
#include<string>
#include<math.h>
#include<vector>
#include<utility>
#include<map>

using namespace std;

int manhatanDistance(pair<int, int> a, pair<int, int> b){
    return abs(a.first - b.first) + abs(a.second - b.second);
}

int main(){
    int n, m;
    int color;
    map<int, vector<int>> colorLists;

    // reading input
    cin >> n >> m;
    int table[n][m]; 

    for(int r = 0; r < n; r++) {
        for (int c = 0; c < m; c++) {
            cin >> table[r][c];
        }
    }

    for (int r = 0; r < n; r++) {
        for (int c = 0; c < m; c++) {
            color = table[r][c]; 
            colorLists[color] = vector<int>();
        }
    }

}

File: codeforces/1600/1648/A.py
from collections import Counter, defaultdict

# reading input
n, m = [int(el) for el in input().split()]

def manhatan_distance(a, b):
    return abs(a[0] - b[0]) + abs(a[1] - b[1])

def compute_manhatan_for_all_table(table, r, c):
    sum_this_number = 0
    this_number = table[r][c]
    for i in range(n):
        for j in range(m):
            if this_number == table[i][j]:   
                sum_this_number += manhatan_distance((r, c), (i, j)) 
    return sum_this_number

table = []
for r in range(n): 
    table.append([int(el) for el in input().split()])

manhatan_distances_dict = defaultdict(int) 
for r in range(n):
    for c in range(m):
        this_number = table[r][c]
        manhatan_distances_dict[this_number] += compute_manhatan_for_all_table(table, r, c)

print(sum(manhatan_distances_dict.values())//2)
    